### PR TITLE
fix: aop `autotest_cookie_keep_before` plugin merge all cookies from report

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
aop `autotest_cookie_keep_before` plugin merge all cookies from report

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJhc3NpZ25lZSI6WyIxMDAxMjA1Il19&id=325656&iterationID=1321&pId=0&type=BUG)

#### Specified Reviewers:

/assign @your-reviewer

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |   aop `autotest_cookie_keep_before` plugin merge all cookies from report           |
| 🇨🇳 中文    |     修复了自动化测试缺失cookie的问题         |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: internal/tools/pipeline/aop/plugins/task/autotest_cookie_keep_before/cookie.go [hunk 1]
```diff
@@ -76,3 +81,19 @@ func appendOrReplaceSetCookiesToCookie(setCookies []string, originCookie string)
...
 	return newCookieStr
 }
+
+func (p *provider) getCookiesFromMeta(meta *structpb.Struct) ([]string, error) {
+	var cookies []string
+	for field, value := range meta.Fields {
+		if field == apitestsv2.HeaderSetCookie {
+			cookieJsonStr := value.GetStringValue()
+			if len(cookieJsonStr) == 0 {
+				return cookies, nil
```
Comment (sfwn) [inline internal/tools/pipeline/aop/plugins/task/autotest_cookie_keep_before/cookie.go:91]: how about `continue` if there are multiple headerSetCookie?

File: internal/tools/pipeline/aop/plugins/task/autotest_cookie_keep_before/plugin.go [hunk 2]
```diff
@@ -57,28 +56,20 @@ func (p *provider) Handle(ctx *aoptypes.TuneContext) error {
...
+	var setCookies []string
 	for _, v := range reportSets.Data.Reports {
 		if v.Meta == nil {
 			continue
 		}
-		meta := v.Meta.AsMap()
-		if meta[apitestsv2.HeaderSetCookie] == nil {
+		cookies, err := p.getCookiesFromMeta(v.Meta)
+		if err != nil {
+			rlog.TErrorf(ctx.SDK.Pipeline.ID, ctx.SDK.Task.ID, "failed to get cookies from meta(continue next report), err: %, meta: %v", err, v.Meta)
```
Comment (sfwn) [inline internal/tools/pipeline/aop/plugins/task/autotest_cookie_keep_before/plugin.go:66]: return error here

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/tools/pipeline/aop/plugins/task/autotest_cookie_keep_before/cookie.go (hunk 1 above)
Comment (chengjoey) [reply internal/tools/pipeline/aop/plugins/task/autotest_cookie_keep_before/cookie.go:91]: adjusted, break if field == apitestsv2.HeaderSetCookie

File: internal/tools/pipeline/aop/plugins/task/autotest_cookie_keep_before/plugin.go (hunk 2 above)
Comment (chengjoey) [reply internal/tools/pipeline/aop/plugins/task/autotest_cookie_keep_before/plugin.go:66]: adjusted

Comment (sfwn) [reply internal/tools/pipeline/aop/plugins/task/autotest_cookie_keep_before/plugin.go:66]: what is `continue next report`?

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/tools/pipeline/aop/plugins/task/autotest_cookie_keep_before/plugin.go (hunk 2 above)
Comment (chengjoey) [reply internal/tools/pipeline/aop/plugins/task/autotest_cookie_keep_before/plugin.go]: deleted `continue next report`

**Review by sfwn — APPROVED** [review]

**Pull request merged**